connector_clickup/models/clickup_backend/common.py

```diff
@@ -0,0 +1,442 @@
+    #         #         ]:
+    #         #             from_date = from_date.strftime("%Y-%m-%dT%H:%M:%SZ")
+    #         #         else:
+    #         #             from_date = to_iso_datetime(from_date)
+    #         #         search_dict.update({"updated": [{"operator": ">", "value": from_date}]})
+
+    #         if model in [
+    #             "clickup.project.project",
+    #             "clickup.project.tasks",
+    #             "clickup.project.task.type",
+    #         ]:
+    #             to_date = import_start_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+    #         else:
+    #             to_date = to_iso_datetime(import_start_time)
+
+    #         if not search_dict.get("updated"):
+    #             search_dict["updated"] = []
+
+    #         search_dict["updated"].append(
+    #             {"operator": "<", "value": to_date, "action": "import"}
+    #         )
+
+    #         filters["search"] = json.dumps(search_dict)
+    #         filters["limit"] = backend.limit
+    #         filters["with_count"] = "true"
+    #         force = False
+    #         if force_update_field:
+    #             force = backend[force_update_field]
+    #         if model == "clickup.project.project":
+    #             job_options["description"] = "Prepare jobs for clickup Project import"
+    #         if model == "clickup.project.tasks":
+    #             job_options["description"] = "Prepare jobs for clickup Task import"
+    #         if model == "clickup.project.task.type":
+    #             job_options["description"] = "Prepare jobs for clickup Stage import"
+    #         clickup_model = (
+    #             self.env[model].with_delay(**job_options or {})
+    #             if with_delay
+    #             else self.env[model]
+    #         )
+
+    #         clickup_model.import_batch(
+    #             backend,
+    #             filters=filters,
+    #             force=force,
+    #             **{"no_delay": not with_delay},
+    #             job_options=job_options,
+    #         )
+    #         if force:
+    #             backend[force_update_field] = False
+    #     next_time = import_start_time - timedelta(seconds=IMPORT_DELTA_BUFFER)
+    #     if from_date_field:
+    #         self.write({from_date_field: next_time})
+
+    def _import_from_date(
+        self,
+        model,
+        from_date_field=None,
+        filters=None,
+        force_update_field=None,
+        priority=None,
+        with_delay=True,
+    ):
+        """Common method for import data from from_date."""
+        if not filters:
+            filters = {}
+        import_start_time = datetime.now()
+        job_options = {}
+        if priority or priority == 0:
+            job_options["priority"] = priority
+        for backend in self:
+            from_date = None
+            if from_date_field:
+                from_date = backend[from_date_field]
+            search_dict = filters.get("search", {})
```

Comment:
Do we need search_dict, also this method needs to be refactored, as a lots of logic is implemented which is of no use.